install.sh

```diff
@@ -269,9 +269,20 @@ check_path() {
 	if gum confirm "create /usr/local/bin/tea?" --affirmative="make symlink" --negative="skip"
 	then
 		echo  #spacer
-		sudo mkdir -p /usr/local/bin
-		sudo ln -sf "$tea" /usr/local/bin/tea
 
+		if [ "$USER" = "root" ] || which sudo >/dev/null;
```

Comment:
it this enough? Can the user be named otherwise? IIRC user-id 0 is always “root” can we check that?

Reply:
Yess, It can be resolved by using `$EUID` instead of `$USER` and changing the condition to:
```
if [ "$EUID" -eq 0 ]
```

Reply:
Technically, we don't care if they're root, but only if they have write access to the directory (or one of its parents, if we have to mkdir). In practice this means root, but it might not if I `sudo chown -R jacob /usr/local`.

Reply:
We only need the write access and the user need not be a root. So, the changes would be ..
```
if [ -w /usr/local ]
```
So, the code would finally look like
```
if [ -w /usr/local ];
then
	mkdir -p /usr/local/bin
	ln -sf "$tea" /usr/local/bin/tea
else
	if which sudo >/dev/null;
	then
		sudo mkdir -p /usr/local/bin
		sudo ln -sf "$tea" /usr/local/bin/tea
	else
		echo  #spacer
		gum format -- <<-EOMD
			> hmmm, sudo command not found.
                        > try installing sudo
			EOMD
	fi
fi
```

Reply:
This should work fine. I think to deal with all the use cases, the logic might be something like this:

if write(/u/l/b) or (write(/u/l) and not(/u/l/b)) or (write (/u) and not(/u/l)) or (write(/) and not(/u)) then mkdir and ln
elif sudo then mkdir and ln
else report an issue

Reply:
I think there should be only a `write(/u/l/b)` because suppose the parent directory is owned by `user`, the child directory  may or may not be owned by root. So, `write(/u/l/b/)` is must.

Reply:
but if it doesn't exist, then writing /usr/local is sufficient. and if that doesn't exist, writing /usr is sufficient. etc. that's the issue.

Reply:
perhaps the logical is is `if -w /usr/local/bin || mkdir -p /usr/local/bin`

Reply:
Makes sense! Addressing it shortly.

Reply:
We should add a stanza to `ci.yml` that tests something like:
```sh
sudo su - root
dpkg -P sudo
$GITHUB_WORKSPACE/install.sh
```
to ensure it functions like we think.

Reply:
Should the job be like:
```
root:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Run as root
        run: sudo su - root && dpkg -P sudo
      - name: Installation
        run: $GITHUB_WORKSPACE/install.sh
```

Reply:
Done!